rallyrolebot/data.py

```diff
@@ -1,142 +1,123 @@
-    table.upsert(
-        {
-            GUILD_ID_KEY: guild_id,
-            COIN_KIND_KEY: coin,
-            REQUIRED_BALANCE_KEY: required_balance,
-            CHANNEL_NAME_KEY: channel,
-        },
-        [GUILD_ID_KEY, CHANNEL_NAME_KEY],
-    )
-
-
-def get_role_mappings(guild_id, coin=None, required_balance=None, role=None):
-    global db
-    table = db[ROLE_MAPPINGS_TABLE]
-    filtered_mappings = table.find(guildId=guild_id)
-    if coin is not None:
-        filtered_mappings = [m for m in filtered_mappings if m[COIN_KIND_KEY] == coin]
-    if required_balance is not None:
-        filtered_mappings = [
-            m for m in filtered_mappings if m[REQUIRED_BALANCE_KEY] == required_balance
-        ]
-    if role is not None:
-        filtered_mappings = [m for m in filtered_mappings if m[ROLE_NAME_KEY] == role]
-    return filtered_mappings
-
-
-def get_channel_mappings(guild_id, coin=None, required_balance=None, channel=None):
-    global db
-    table = db[CHANNEL_MAPPINGS_TABLE]
-    filtered_mappings = table.find(guildId=guild_id)
-    if coin is not None:
-        filtered_mappings = [m for m in filtered_mappings if m[COIN_KIND_KEY] == coin]
-    if required_balance is not None:
-        filtered_mappings = [
-            m for m in filtered_mappings if m[REQUIRED_BALANCE_KEY] == required_balance
-        ]
-    if channel is not None:
-        filtered_mappings = [
-            m for m in filtered_mappings if m[CHANNEL_NAME_KEY] == channel
-        ]
-    return filtered_mappings
-
-
-def remove_role_mapping(guild_id, coin, required_balance, role):
-    global db
-    table = db[ROLE_MAPPINGS_TABLE]
-    table.delete(
-        guildId=guild_id, coinKind=coin, requiredBalance=required_balance, roleName=role
-    )
-
-
-def remove_channel_mapping(guild_id, coin, required_balance, channel):
-    global db
-    table = db[CHANNEL_MAPPINGS_TABLE]
-    table.delete(
-        guildId=guild_id,
-        coinKind=coin,
-        requiredBalance=required_balance,
-        channel=channel,
-    )
-
-
-def add_discord_rally_mapping(discord_id, rally_id):
-    global db
-    table = db[RALLY_CONNECTIONS_TABLE]
-    table.upsert({DISCORD_ID_KEY: discord_id, RALLY_ID_KEY: rally_id}, [DISCORD_ID_KEY])
-
-
-def get_rally_id(discord_id):
-    global db
-    table = db[RALLY_CONNECTIONS_TABLE]
-    row = table.find_one(discordId=discord_id)
-    if row is not None:
-        return row[RALLY_ID_KEY]
-    return None
+import dataset
+
+import config
+from constants import *
+
+db = None
+
+"""Functions for managing a dataset SQL database
+    # Schemas
+
+    #################### mappings ######################
+    guildId
+    coinKind
+    requiredBalance
+    roleName
+
+    #################### channel_mappings ######################
+    guildId
+    coinKind
+    requiredBalance
+    channelName
+
+    #################### rally_connections ######################
+    discordId
+    rallyId
+"""
+def connect_db():
+    global db
```

Comment:
Please avoid using global variables.

Reply:
Makes sense, I didn't touch this part of Calvin's code yet. I think the typical thing to do is establish a new database connection, make changes, and then close it, which should eliminate the need for any global database object.

Reply:
That sounds like a better approach, would you mind making that shift Ashneal? And should we make that a separate branch/PR once we have landed this?

Reply:
Sounds like a plan - I'll make those changes. Since no new pull requests have been accepted other than updating to Calvin's latest code, I will simply continue committing to this pull request.